v3/src/data-interactive/handlers/color-map-handler.ts

```diff
@@ -0,0 +1,87 @@
+    const categorySet = createProvisionalCategorySet(dataContext, attributeId)
+    metadata.promoteProvisionalCategorySet(categorySet)
+    if (!categorySet) return errorResult("Category set not found")
+    Object.entries(createColorMap || {}).forEach(([category, color]) => {
+      categorySet.setColorForCategory(category, color ?? categoryColorMap[category] ?? "")
+    })
+    categorySet.storeAllCurrentColors()
+    metadata.promoteProvisionalCategorySet(categorySet)
+    const categoryColorMap: Record<string, string | undefined> = categorySet?.colorMap || {}
+    if (!categoryColorMap) return errorResult("Category color map not found")
+
+    return {
+      success: true,
+      values: categoryColorMap
+    }
+
+  },
+
+  get(resources: DIResources, _values?: DIValues & { metadata?: any }) {
+    const { dataContext, collection, attribute } = resources
+    if (!dataContext) return dataContextNotFoundResult
+    if (!attribute) return attributeNotFoundResult
+    if (!collection) return collectionNotFoundResult
+
+    const metadata = getSharedCaseMetadataFromDataset(dataContext) ?? _values?.metadata
+    if (!metadata) return errorResult("Metadata not found")
+    const attributeId = attribute.id
+    const categorySet = metadata.getCategorySet(attributeId)
+    const categoryColorMap: Record<string, string | undefined> = categorySet?.colorMap || {}
+    if (!categoryColorMap) return errorResult("Category color map not found")
+
+    return {
+      success: true,
+      values: categoryColorMap
+    }
+  },
+
+  update(resources: DIResources, _values?: DIValues & { metadata?: any }) {
+    const { dataContext, collection, attribute } = resources
+    if (!dataContext) return dataContextNotFoundResult
+
+    const metadata = getSharedCaseMetadataFromDataset(dataContext) ?? _values?.metadata
+    if (!metadata) return errorResult("Metadata not found")
+    if (!collection) return collectionNotFoundResult
+    const attributeId = attribute && typeof attribute === 'object' ? attribute.id : undefined
+    if (!attributeId) return attributeNotFoundResult
+    const newColorMap = (_values && 'colorMap' in _values) ? _values.colorMap as DICategoryColorMap : undefined
+    if (!newColorMap) return errorResult("New color map not found")
+    const categorySet = metadata.getCategorySet(attributeId)
+    if (!categorySet) return errorResult("Category set not found")
+      const categoryColorMap: Record<string, string | undefined> = categorySet?.colorMap || {}
+    if (!categoryColorMap) return errorResult("Category color map not found")
+
+    Object.entries(newColorMap || {}).forEach(([category, color]) => {
+      categorySet.setColorForCategory(category, color ?? categoryColorMap[category] ?? "")
+    })
+    categorySet.storeAllCurrentColors()
+    const updatedColorMap = categorySet.colorMap
+    if (!updatedColorMap) return errorResult("Updated color map not found")
+    return {
+      success: true,
+      values: updatedColorMap
+    }
```

Comment:
There are lots of similarities between this code and the code for `create`, so please apply relevant suggestions for that handler here as well.